backend/Makefile

```
@@ -0,0 +1,139 @@
+ifeq ($(PYTHON_VERSION_OK),0)
+  $(error "Need python $(PYTHON_VERSION) >= $(PYTHON_VERSION_MIN)")
+endif
+
+PLONE_SITE_ID=Plone
+BACKEND_FOLDER=$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+PLONE_VERSION=$(shell cat $(BACKEND_FOLDER)/version.txt)
+EXAMPLE_CONTENT_FOLDER=${BACKEND_FOLDER}/src/collective/addon/setuphandlers/examplecontent
+
+GIT_FOLDER=$(BACKEND_FOLDER)/.git
+VENV_FOLDER=$(BACKEND_FOLDER)/.venv
+BIN_FOLDER=$(VENV_FOLDER)/bin
+
+
+all: build
+
+# Add the following 'help' target to your Makefile
+# And add help text after each target name starting with '\#\#'
+.PHONY: help
+help: ## This help message
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+$(BIN_FOLDER)/pip $(BIN_FOLDER)/tox $(BIN_FOLDER)/pipx $(BIN_FOLDER)/uv $(BIN_FOLDER)/mxdev:
+	@echo "$(GREEN)==> Setup Virtual Env$(RESET)"
+	$(PYTHON) -m venv $(VENV_FOLDER)
+	$(BIN_FOLDER)/pip install -U "pip" "uv" "wheel" "pipx" "mxdev" "tox" "pre-commit"
+	if [ -d $(GIT_FOLDER) ]; then $(BIN_FOLDER)/pre-commit install; else echo "$(RED) Not installing pre-commit$(RESET)";fi
+
+instance/etc/zope.ini: $(BIN_FOLDER)/pip  ## Create instance configuration
+	@echo "$(GREEN)==> Create instance configuration$(RESET)"
+	$(BIN_FOLDER)/pipx run cookiecutter -f --no-input --config-file instance.yaml gh:plone/cookiecutter-zope-instance
+
+.PHONY: config
+config: instance/etc/zope.ini
+
+.PHONY: build-dev
+build-dev: config ## Install Plone packages
+	@echo "$(GREEN)==> Setup Build$(RESET)"
+	$(BIN_FOLDER)/mxdev -c mx.ini
+	$(BIN_FOLDER)/uv pip install -r requirements-mxdev.txt
+
+.PHONY: install
+install: build-dev ## Install Plone
+
+.PHONY: build
+build: build-dev ## Install Plone
+
+.PHONY: clean
+clean: ## Clean environment
+	@echo "$(RED)==> Cleaning environment and build$(RESET)"
+	rm -rf $(VENV_FOLDER) pyvenv.cfg .installed.cfg instance .tox .venv .pytest_cache
+
+.PHONY: start
+start: ## Start a Plone instance on localhost:8080
+	PYTHONWARNINGS=ignore $(BIN_FOLDER)/runwsgi instance/etc/zope.ini
+
+.PHONY: console
+console: instance/etc/zope.ini ## Start a console into a Plone instance
+	PYTHONWARNINGS=ignore $(BIN_FOLDER)/zconsole debug instance/etc/zope.conf
+
+.PHONY: create-site
+create-site: instance/etc/zope.ini ## Create a new site from scratch
+	PYTHONWARNINGS=ignore $(BIN_FOLDER)/zconsole run instance/etc/zope.conf ./scripts/create_site.py
+
+# Example Content
+.PHONY: update-example-content
+update-example-content: $(BIN_FOLDER)/tox ## Export example content inside package
+	@echo "$(GREEN)==> Export example content into $(EXAMPLE_CONTENT_FOLDER) $(RESET)"
+	if [ -d $(EXAMPLE_CONTENT_FOLDER)/content ]; then rm -r $(EXAMPLE_CONTENT_FOLDER)/* ;fi
+	$(BIN_FOLDER)/plone-exporter instance/etc/zope.conf $(PLONE_SITE_ID) $(EXAMPLE_CONTENT_FOLDER)
+
+.PHONY: check
```

See https://github.com/plone/cookiecutter-volto/issues/14#issuecomment-2068663318

```suggestion
.PHONY: lint
```

---

@stevepiercy As we are now using pre-commit, it seems better to have just the `check` command instead of `format` and `lint` -- even more when considering the old `format` command had to be run multiple times to format the entire codebase.

---

Ultimately, what does this command do? `check` what? And `check` for what?

Does it write files? If so, then that is not checking, but formatting or linting. If it only analyzes code and reports problems, then that's checking.